### PR TITLE
fix/test/docs: negative interval accumulation with spread rewards

### PR DESCRIPTION
**Pull request by p0mvn**
Progress towards #5854

## What is the purpose of the change

This PR fixes a negative interval accumulation edge case bug by simply replacing `Sub` with `SafeSub` where applicable.

The tests are extended to cover claiming rewards below, inside, and above the position after it is initialized.

Note, that incentive rewards still need to be tested and potentially fixed. This PR only covers spread rewards.

## Testing and Verifying

The global invariants are asserted, and the tests ensure that the appropriate amount is claimed from all positions in all cases.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [X] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [X] Code comments?
  - [ ] N/A

### CODE REVIEW

**Comment by ValarDragon** [comment]
Lgtm on first pass! Will review test changes in more detail later.

File: x/concentrated-liquidity/README.md [hunk 1]
```diff
@@ -1244,6 +1244,76 @@ func (k Keeper) collectSpreadRewards(
...
+
+For this reason, there are 3 ways to compute the rewards accrued inside the position:
+1. current tick is below the position's range (current tick < lower tick)
+- Then, we compute rewards as `lower tick snapshot - upper tick snapshot`
+2. current tick is within the position's range (lower tick <= current tick < upper tick)
+- Then, we compute rewards as `global accumulator - upper tick snapshot - lower tick snapshot`
+3. current tick is above the position's range (current tick >= upper tick)
+- Then, we compute rewards as `upper tick accumulator - lower tick accumulator`
+
+### Negative Interval Accumulation Edge Case Behavior
```
Comment (ValarDragon) [inline x/concentrated-liquidity/README.md:1297]:
```suggestion
### Negative Interval Accumulation Behavior
```

File: x/concentrated-liquidity/invariant_test.go [hunk 2]
```diff
@@ -102,12 +103,18 @@ func (s *KeeperTestSuite) assertTotalRewardsInvariant(expectedGlobalRewardValues
 		totalCollectedIncentives = totalCollectedIncentives.Add(collectedIncentives...)
 	}
 
+	additiveTolerance := sdk.Dec{}
+	if !expectedGlobalRewardValues.ExpectedAdditiveTolerance.IsNil() {
+		additiveTolerance = expectedGlobalRewardValues.ExpectedAdditiveTolerance
+	}
```
Comment (ValarDragon) [inline x/concentrated-liquidity/invariant_test.go:109]: why did we do this?

**Review by ValarDragon — APPROVED** [review]

**Comment by ValarDragon** [comment]
We may have to fix the osmoutils version if E2E fails

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM, good work!

**Pull request merged**